addons/l10n_tr_nilvera/lib/nilvera_client.py

```diff
@@ -0,0 +1,68 @@
+import logging
+import requests
+from datetime import datetime
+from pprint import pformat
+
+from odoo import _
+from odoo.exceptions import UserError, ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class NilveraClient:
+    def __init__(self, environment=None, api_key=None, timeout_limit=None):
+        self.is_production = environment and environment == 'production'
+        self.base_url = 'https://api.nilvera.com' if self.is_production else 'https://apitest.nilvera.com'
+        self.timeout_limit = min(timeout_limit, 30) if timeout_limit else 10
+
+        self.session = requests.Session()
+        self.session.headers.update({'Accept': 'application/json'})
+        if api_key:
```

Comment:
Is it mandatory to have a api_key in prod ? If yes, it might be nice to raise an error if there is none